Trebuchet : Add 2 Hotseat preferences

CmoaToto opened the pull request:
This adds 2 preferences for the hotseat (Launcher main bar)
- Choose the number of icons on the hotseat
- Choose the place of the applications drawer button
- Translations in English + French

This is usefull for big screen devices.
You can set from 1 to 9 buttons on the hotseat.
You can set the app button where you want.
Default values  are keeped unchanged so there is no pb with first screen
tuto + no pb with default shortcuts.

nebkat committed:
Preferences: initial Preferences Activity, Provider and XML

Change-Id: I27d44993c4e34e08cfafff6d7a418df7c4f06fd2

nebkat committed:
Preferences: Hide search bar preference + others

- Restart Launcher on preferences changed
- Implement "getCurrentOrientation()"

Change-Id: I0728824ab994984c1dc798b61efa26b727ca3ea2

nebkat committed:
Preferences: Resize any widget

Change-Id: I7ef5546f57543664d45a25eb5b9d07f303c9db0b

nebkat committed:
Preferences: Separate Apps and Widgets

Reverts "Add ability to swipe from apps to widgets in AllAppsCustomize"

Change-Id: I0727fbe747f2f93c4680bf91fbcbed75af662b29

Danesh committed:
Preferences: Auto-rotate screen

Change-Id: I126b57932bad994c3fcfa3c764d94bdb60037a93

nebkat committed:
Preferences: Show Page Indicator (Workspace)

Change-Id: I564eca3da962f5af088e26b85d13dc3585780068

nebkat committed:
Preferences: Fade Page Indicator (Workspace)

Change-Id: I7c30adaac1ddaea69e409410b53185926a9266f6

nebkat committed:
Preferences Show/Fade Page Indicator (AllApps)

Change-Id: I9f417ecfdfcfdbcf3f4dc5f633878228194faaba

Danesh committed:
Folders: Merge Folders

This allows a folder A's contents to be merged into folder B
when folder A is drag/dropped onto folder B.

Change-Id: Ie9461fa5ccaa02e06b871edd3eb3cf7798070028

Danesh committed:
Preferences: NumberPickerPreference

Change-Id: I1888fe530a71af1e5d3b015d4a416c98e20ee3e0

nebkat committed:
Preferences: Number of Homescreens + Default Screen

Change-Id: Id361d71c9fd0f0748fa6a48c6702178d366221b8

nebkat committed:
Preferences: Scroll Wallpaper

Change-Id: If20b0f45cc773f7e03565b638ae745ced3070d5c

nebkat committed:
Preferences: Cleanup

Change-Id: I1c68f8b58025714138c0c6bf0e1faf1f01d17be9

nebkat committed:
Tweak Menu

- Move "System settings" to the bottom of the menu - Request from Pawit Pornkitprasan
- Change "Preferences" to "Trebuchet settings" - Request from Tanguy Pruvot
- Hide Preferences if using CyanogenMod (with the settings shortcut) - Request from Steve Kondik

Change-Id: I0738fb0feb04d601b93e28088b771283c35b6684

nebkat committed:
Update Copyright

Change-Id: I5ffdfd4a09bd58f8ae498c932ce07d691cfb8ab8

nebkat committed:
AppsCustomize: Apps Sorting Mode

Change-Id: I9b376c39da3039b7c9a37f0748a1d6b5dc67a471

Commit:
AppsCustomize: Apps Sort Mode Cling

Change-Id: I07a08afb1f249f226693e935ad7ebc017f9c79b0

Commit:
AppWidgetResizeFrame: Allow resizing to any size

Widgets normally aren't allowed to get resized to less than their
original size. This enables it when the "Resize Any Widget" pref
is enabld.

Change-Id: I5cb94b598fe31f2bfa7c445aa0207c781a4cad6d

Commit:
Preferences: Show Dock Divider

Change-Id: I5e4349e5b7539a66d5d96c5f9fd9780b7572d624

Commit:
Preferences: Version preference information style

Change-Id: I723b6daf3e6a8f1d2cc2c1c5f4f50b85ef7ecd08

Commit:
DeleteDropTarget: Long-Hold To Uninstall

Change-Id: I2f60040c4cc645a54e9a8e696f330a8e70ff833c

kladd committed:
Preferences: Show/hide homescreen icon labels

Change-Id: I4057152c571d73e4da2f1a43454feb152c0249af

Commit:
CellLayout: Better Scaling

Change-Id: I49eb2bbb9d49e5bfba1337d99a246fbceb61f86e

Commit:
DeleteDropTarget: "Cancel" when dragging from AllApps

Change-Id: I4fb03b0cbffe9662bcf6c582fe5bfb5bbbcac20e

Commit:
DeleteDropTarget: Haptic Feedback on Long-Press

Change-Id: I37ec1e3ab8df5c144b777776323faad8c0db1273

Commit:
Workspace: Custom Vertical/Horizontal Padding

Change-Id: Ie9b74b081f254493905ab792bfe85f95a6641a79

Commit:
DropTarget: Manage landscape strings programmatically

Change-Id: I92bba3aa05e2ba0de9fb29dbfd38a33ebe4d0c4a

Commit:
Workspace: Cleanup hide icon labels

Change-Id: I3268f9a0a2187b0f7ff32a97d1566e04a454657b

Commit:
NumberPickerPreference: Cleanup external min/max code

Change-Id: I5f67a1e382cd96852836f5a2d5f79affd5f16abe

Commit:
Preferences: Hide dock/icon categories

Change-Id: I6ea5dd6c3433dd5be73f5f332fadaf8deade522f

Commit:
DoubleNumberPickerPreference

Change-Id: I9e71b1f889e98e134b99c965452192e503ff2f08

Commit:
Preferences: Grid Size

Change-Id: I51bec0a0c29421fc389f7b656496ba1467758892

Commit:
Hotseat: Default Cell Counts

Change-Id: I9aad8d04145bd20ae7b2b6198c556254f949aba4

Commit:
PagedView: Vertical

Change-Id: Ifed3f713da1f482e67d22765a6215e7d194194eb

Conflicts:
	src/com/cyanogenmod/trebuchet/Workspace.java

Commit:
AppsCustomizePagedView: Orientation

Conflicts:
	res/values/arrays.xml
	src/com/beansoft/launcher/preference/PreferencesProvider.java
	src/com/cyanogenmod/trebuchet/AppsCustomizePagedView.java

Change-Id: I21070fb3688ba436b3ca88f1caee8a6cfbdbf31d

Commit:
Launcher: Use correct toolbar icon metadata name

Change-Id: Ic1ef15c0216122fce80db21b2ca8488293fcae2e

Commit:
CM Preferences Category

Change-Id: I67a5f9048a182e9e5235423cfa4c7a69a99debed

Commit:
Launcher: increasing the lower bound and upperbound of loaded pages to avoid stutter in applist scroll

Change-Id: I8fed59569129cc1c062ffdb287183531c1f84dbf
(cherry picked from commit 4cc5af01944981714dea73de6556f2db0456b621)

jruesga committed:
Fix Issue 5810: Trebuchet: Uighur language

Initial commits for Uighur language
Depends: http://r.cyanogenmod.com/#/c/19818/1
Credits: Abduqadir Abliz

Patch 2: Remove trailing whitespaces

Change-Id: Ia7c88fbc39f843b3c834291a7a077e8b2bd1a23c

Commit:
Ensuring that restoreInstanceState is being called promptly for synchronously bound page

Change-Id: I0e71c29f553ad360ec42a6a0b2529d16cbddd437

chirayudesai committed:
Rename Launcher to Trebuchet
Launcher2 is now Trebuchet
application_name removed from localized strings and made un-translatable
com.android.launcher is now com.cyanogenmod.trebuchet

tothtamassyman committed:
Added hungarian translation for Trebuchet

Change-Id: I0bc00ac123fa5d00f2fd8269db27d9ba8ed8e12b

Commit:
Update Russian Translation-Trebuchet-CM10

Update Russian Translation-Trebuchet-CM10

Update Russian Translation-Trebuchet-CM10

Update Russian Translation-Trebuchet-CM10

Update Russian Translation-Trebuchet-CM10

Update Russian Translation-Trebuchet-CM10

codeworkx committed:
values-sw: application_name is not translatable

Change-Id: Id5477e35b7cdd61a5f0114368d2b61e065a6471e

Gustav0ar committed:
PT-BR: Merge From CM9 Translations

Change-Id: Id3c9bd033b5c44dc5ab6edb41c9455ef17af5a3d

Jiangyi committed:
Trebuchet: Simplified Chinese Translations

Ported from CM9.

Change-Id: Iad7b368c15077ca15a72953e442eee153c214846

blunden committed:
Use correct search and voice search icons in the search bar

Change-Id: Ie146b7945a9c0226f95563be3f593d467a2a00c3

Fra-90 committed:
Trebuchet - Updated Italian translation

Change-Id: I24a6f418ac175b6bf2424e5b1c5b5b9e1133c621

Rashed97 committed:
New App Icon featuring Cid

Changed app icon to include Cid instead of the default Android
Also added .gitignore for .DS_Store files
Patch set 1: Initial Commit
Patch set 2: Added .gitignore, added minor inner and drop shadows to Cid to make
it look nicer
Patch set 3: Added minor darkeness around Cid to counter the light around him
Also made minor enhancements to the overall graphic for smoothness and
eveness with smooth lighting
Patch set 4: Removed .gitignore
Patch set 5: Clean dependancies

Change-Id: I8c9ceeb763534e3f6351e2861231e0c1ee249824

jruesga committed:
Trebuchet: Uighur - Remove obsolete translation entries

Remove obsolete translation entries

Change-Id: I2eac3f954857ca58912ec7b78fd49006e6dc239c

niko0o committed:
trebuchet: updated finnish translations

Change-Id: I3e66fe6942d1658b22809f192ac9781ee374f179

Danesh committed:
Trebuchet : Hide voiceSearch icon

Hide the actual voice search icon when hiding the bar.

Change-Id: I51b700ba88ddd19006f7eb0d0d572f255450413a

Gustav0ar committed:
PT-BR: Merge more from CM9 translations

Change-Id: I96baf11d4a6cf6e43c3eeabdfb3e7a75e8d0bdfa

Commit:
Update Russian Translation-Trebuchet-CM10

elektroschmock committed:
Trebuchet: German translation

Patch set 1 - Add missing strings and rearrange
Patch set 2 - Add some more missing strings

Change-Id: I29e0f26c65bf067ef856515b468bf8ea5dd0af0a

Cbienmoua committed:
Update French translation-Trebuchet-CM10

Change-Id: I5b29a68b4e7e849c3948da032f6aba6b2274ecaa

takuo committed:
Trebuchet: Update Japanese translation.

Change-Id: I7bcf7c60d5f0751face21d06c8b5b9d97726630b

therbom committed:
Trebuchet: Dutch translations

Change-Id: Iae592699d492667624431742792f82c15bebc70a

Fidoble committed:
Trebuchet:
Added missing danish translations

Change-Id: I0e9f54e6fc356dec1ed9e19bef81559ac7bafdce

Commit:
Ensuring that unbind calls are not queued in the worker thread and executed subsequently on the main thread.  (Bug 7001531)

Change-Id: I677242c0952d6385ca229ba300fdd5a2a7908a8a

Commit:
Merge "Fixing a couple small issues with reordering (issue 7064138), (issue 7064337)" into jb-mr1-dev

Commit:
merge in jb-mr1-release history after reset to jb-mr1-dev

Commit:
Merge "Ensuring that unbind calls are not queued in the worker thread and executed subsequently on the main thread.  (Bug 7001531)" into jb-mr1-dev

Commit:
Fixing some spacing issues related to new 10inch layout. (Bug 6955380, Bug 7035262, Bug 7056508)

Change-Id: I589fe6c2475cfff12599eee9b490b2d4dc86511a

Commit:
merge in jb-mr1-release history after reset to jb-mr1-dev

Commit:
Fix an error that is not being expressed

Change-Id: I8c3488906b5b95aaf20b996f488ff01bad76300c

Commit:
Fixing issue where we were not removing items from LauncherModel and only the add-queue. (Bug 7078244)

Change-Id: I5660b447521646f820fb2497f327e965d99ac207

Commit:
es_ES: Added missing translations

Change-Id: I93469ff854fc5c5750dec0c2964c4a5a3078366f

Jacob-DK committed:
Trebuchet: Suggested minor improvements to Danish translation

Change-Id: Ibf5f31d60a33faebd04ad048d72dac9ee1d63b82

mikejurka committed:
Adding more debug code for dup. ItemInfo bug

Change-Id: I7963a59b54f05e29e485c3dcc5c0e159e36de59d

mikejurka committed:
Adding more debug code for dup. ItemInfo bug

Change-Id: I7963a59b54f05e29e485c3dcc5c0e159e36de59d

mikejurka committed:
Debug code for bug b/6830207

Change-Id: Ic951fd448a74ac84398f3e4371b2410e295f0293

ygorigor committed:
Trebuchet: update romanian translation

Change-Id: Ib16ae7333bd3b5d98d54c9c9db7bb52cf255457b
Signed-off-by: Igor <igor_sorocean@hotmail.com>

mikejurka committed:
Merge "Debug code for bug b/6830207" into jb-mr1-dev

Commit:
merge in jb-mr1-release history after reset to jb-mr1-dev

burnsra committed:
Fix for install shortcut intent

- Enables the ability to install a shortcut to the home screen

Change-Id: I6e3e6c36d40907082ea1f3bf11f30110786f8ce9

Commit:
Merge "Fixing issue where we were not removing items from LauncherModel and only the add-queue. (Bug 7078244)" into jb-mr1-dev

Commit:
merge in jb-mr1-release history after reset to jb-mr1-dev

mikejurka committed:
Merge "Adding more debug code for dup. ItemInfo bug" into jb-mr1-dev

rheeze committed:
Trebuchet: Updated italian translation

Change-Id: Ic899c654021afa145228564d99cb64de520aa33f

Jiangyi committed:
Trebuchet: Simplified Chinese Translations

Ported from CM9 with a slight modification.

Change-Id: I960e7c769d4bf586c8161e44cc58815214782017

Commit:
Workaround for generating QSB icon press states. (Bug 7132477)

Change-Id: I7754bbb20572908cd428c12ba84a7b1160063bcb

Commit:
Workaround for generating QSB icon press states. (Bug 7132477)

Change-Id: I7754bbb20572908cd428c12ba84a7b1160063bcb

Commit:
Making launcher use new widget binding APIs

Change-Id: I9cd6716e1dc0c53b5c846371ea109ced4cd3d40a

burnsra committed:
Fix shortcut intent when using less than 3 homescreens

This patch fixes issue 6192 as outlined at the URL below:
http://code.google.com/p/cyanogenmod/issues/detail?id=6192

To reproduce the issue set the number of home screens to 2 or less and try to add a shortcut as follows:
- open "People" application
- select a contact
- using context menu, choose "Place on Home screen"
- go to home screen to see crash

Change-Id: Ib1c7f6b1cf5055714b4d9e57c637c8a7fa05e9d8

AndroidMeda committed:
Trebuchet/TR: Initial commit for Turkish

Change-Id: I8ec08114ba8663a48214de7ecd9ecac68623ff7b

Commit:
Add user initialing receiver to select a unique wallpaper.

Change-Id: I14652d8108c2e19db821ab5526b6e630db8fe998

Commit:
merge in jb-mr1-release history after reset to jb-mr1-dev

Commit:
Fixing issue where we were collating labels based on an old Locale. (Bug 7133873)

Change-Id: Ic1366f06020d1e405a6800386472a8832d683c5e

mikejurka committed:
Fixing some warnings

Change-Id: I28ff3b62b831bd49e3592c1b86ba978208324c11

Commit:
merge in jb-mr1-release history after reset to jb-mr1-dev

mikejurka committed:
Merge "Fixing some warnings" into jb-mr1-dev

Commit:
Remove wallpaper offset

TBH, it seems to do nothing, only breaks themeability of tablet statusbar
(black line when system bar opacity is > 100)

Change-Id: If0a5556b49eb8c50851b0630b5d0de56d8c8b687

Commit:
merge in jb-mr1-release history after reset to jb-mr1-dev

Commit:
merge in jb-mr1-release history after reset to jb-mr1-dev

Commit:
Removing functionality to click under AllApps to return to the workspace. (Bug 7012020)

Change-Id: I69a4d28866d550cb4a43c99875e3e94dda1b0de8

narayank committed:
Remove "fade in" animation for search + voice search.

This applies when the app is launched by tapping the
search box or the voice search icon.

bug:7193408
Change-Id: I29f41faa20c3e7ad00032f749c99d9b69402c422

mikejurka committed:
Merge "Remove "fade in" animation for search + voice search." into jb-mr1-dev

mikejurka committed:
Try to fix RTE when updating shortcuts

Bug: 6529398
Change-Id: Ib88eda5dc1eaf72aae161b25b080197b1049e824

mikejurka committed:
Merge "Try to fix RTE when updating shortcuts" into jb-mr1-dev

mikejurka committed:
Fix NPE

Bug: 7024933

Commit:
merge in jb-mr1-release history after reset to jb-mr1-dev

burnsra committed:
Fix shortcut intent when all homescreens are full

This patch fixes issue 6218 as outlined at the URL below:
http://code.google.com/p/cyanogenmod/issues/detail?id=6218

To reproduce the issue completely fill all available homescreens and try to add a shortcut as follows:
- open "People" application
- select a contact
- using context menu, choose "Place on Home screen"
- go to home screen to see crash

Change-Id: Iab43db930a3fb305cae4baaa24100d5d16d7b96b

Commit:
7225161 Launcher needs to add android:supportsRtl="true" in his AndroidManifest

- add basic RTL support in the AndroidManifest so that Widgets will be able
to support RTL too
- force LTR layout direction to AppsCustomizeTabHost so that we dont have to
deal with layout mirroring yet

(do not need to force also LTR layout direction to Wallpaper activity and fragment
as another CL is removing the ActionBar on them)

Change-Id: I66623a3fcec0d7adef62b0ab8c8b2170e09f1b1b

mikejurka committed:
Remove action bar from wallpaper picker

Bug: 5295678

Change-Id: I50bb4bee3d9d54f28ef4a46ed9754701045d38c7

Commit:
Merge "7225161 Launcher needs to add android:supportsRtl="true" in his AndroidManifest" into jb-mr1-dev

mikejurka committed:
Fix # 6925851: Workspace page spacing gets messed up

Change-Id: I694fa1e04049e35a415b1f1a85759e501fec6258

Commit:
merge in jb-mr1-release history after reset to jb-mr1-dev

mikejurka committed:
Fix 7223056: Enable scroll indicator on tablets

Commit:
merge in jb-mr1-release history after reset to jb-mr1-dev

Commit:
Updating launcher layouts (Bug 6955380)

- Fixing issue with tiny search/voice icons in landscape

Change-Id: I0291c3415bda0aece241fa07cad132de270d1dda

Commit:
Merge "Updating launcher layouts (Bug 6955380)" into jb-mr1-dev

Commit:
Fixing typo in Mako resource bucket (Bug 6955380)

Change-Id: I85b2db135ab8bdb4a349c070811532c877201191

FaultException committed:
Trebuchet: Only mark preferences changed when they actually are

Otherwise a redraw is forced by just entering the launcher
settings.

Change-Id: Iafbd20368650bc4ea751976892d7184248937118

Commit:
merge in jb-mr1-release history after reset to jb-mr1-dev

Commit:
Fix issue #7255954: API Review: rename Dream to DreamService

Change-Id: I4a8fcea6856ea021b22d6e1f317c43d081c7b923

Commit:
merge in jb-mr1-release history after reset to jb-mr1-dev

mikejurka committed:
Fix issue where clicking home button caused search icon blip

Also, tweak size of icons in landscape mode

Bug: 7221345

Change-Id: Ib0d1445e165f291c09021f7dad08ad4d248e47ca

Commit:
Specify workspace resource in preload broadcast

Bug 7203884

Allow specifying the workspace XML layout in the PRELOAD_WORKSPACE
broadcast.

Change-Id: Iec01c6fa2dde4635f624f040b0772ee11fcd88dc

mikejurka committed:
Fix size of QSB icons in landscape on sw600dp devices

Bug: 7260607
Change-Id: I8609449ed1a927a213f2f3d4f2c32c99f24f12d2

Commit:
merge in jb-mr1-release history after reset to jb-mr1-dev

Commit:
Merge "Specify workspace resource in preload broadcast" into jb-mr1-dev

FaultException committed:
Trebuchet: Fix preference casing

Change-Id: I62bd7c9bb1906b9fb09be7a577d7a58e28aa08ab

Commit:
merge in jb-mr1-release history after reset to jb-mr1-dev

mikejurka committed:
Fix bug 6886990: cling accessibility

Clings should block interaction with elements
below when Talkback is enabled

Change-Id: Ifdcd109351fd3c2d6dc97ad75ea3ab6eea48b582

Commit:
Fix for widgets with config not getting size info (issue 7266053)

Change-Id: I0b55555ed38d4f2a70345d1b6316a075a9346111

Commit:
Removing db corruption exceptions, logging instead (issue 7112132)

Change-Id: I5fd169ba53a88797723997d25399b39b0468864a

mikejurka committed:
Add support for custom manta cling

- Load a custom cling if custom workspace layout is specified
- Also persist any custom workspace layout

Bug: 6992663

Commit:
merge in jb-mr1-release history after reset to jb-mr1-dev

Commit:
Fix to ensure default widgets get size callbacks (issue 7287744)

Change-Id: I46b329db247a0b450493530ab619576c45418b01

jozka1 committed:
Trebuchet: Added czech translations

P3: update

Change-Id: Iabcd338671521737c948e4426e1d4fe80ca8e2a8

Commit:
Genericize naming of sw720dp wallpapers

Bug: 6678111
Bug: 7032938

Change-Id: Ie9cf0f295ac090dd2692f6a1e55d4074be07508c

Commit:
Merge "Fix to ensure default widgets get size callbacks (issue 7287744)" into jb-mr1-dev

Commit:
Merge "Genericize naming of sw720dp wallpapers" into jb-mr1-dev

Commit:
merge in jb-mr1-release history after reset to jb-mr1-dev

mikejurka committed:
Revert "Add support for custom manta cling"

This reverts commit 68255cbabe1cb7fbfc0706a8a2e1b1927eae1b55.

mikejurka committed:
Merge "Revert "Add support for custom manta cling"" into jb-mr1-dev

Commit:
merge in jb-mr1-release history after reset to jb-mr1-dev

mikejurka committed:
Re-merge "Add support for custom manta cling"

Also, fix issue where mako was crashing because
Launcher was trying to load a custom cling for it.
It was doing this because a custom home layout was
specified by SetupWizard, but for now we only want
custom clings on large devices

Bug: 7302705
Bug: 6992663

This reverts commit 2ebcf509f038f435ca8819dc098a7cb480bed0d9.

Change-Id: I46533309d5c62edfbd3a3b178c6965ffc73b6476

mikejurka committed:
Fix bug 7288630: Folder icons clipped on manta

mikejurka committed:
Bug 7301303: more rows in all apps for sw720dp-port

Commit:
Import translations. DO NOT MERGE

Change-Id: I634bec03bc0dea7d6ce2e6d90533ab1c4ad79b7c
Auto-generated-cl: translation import

mikejurka committed:
Merge "Bug 7301303: more rows in all apps for sw720dp-port" into jb-mr1-dev

mikejurka committed:
Merge "Fix bug 7288630: Folder icons clipped on manta" into jb-mr1-dev

mikejurka committed:
Merge "Fix bug 6886990: cling accessibility" into jb-mr1-dev

mikejurka committed:
Merge "Re-merge "Add support for custom manta cling"" into jb-mr1-dev

mikejurka committed:
Stop using mcc for custom workspace cling on sw600dp devices

Bug: 7203884

Commit:
merge in jb-mr1-release history after reset to jb-mr1-dev

Commit:
Tweaking workspace overscroll effect to prevent clipping icons. (Bug 7267045)

Change-Id: I96514f762a1af1ec66523afde7db7a967b625afb

Commit:
merge in jb-mr1-release history after reset to jb-mr1-dev

Commit:
Import translations. DO NOT MERGE

Change-Id: Iac3a250c73e5f0b3b80675a1a993c8336ba6ade9
Auto-generated-cl: translation import

mikejurka committed:
Update custom cling params

Bug: 7318080

Change-Id: I77d73d81a60efa3886f088e82c6ba3466f6ac331

Commit:
Merge "Tweaking workspace overscroll effect to prevent clipping icons. (Bug 7267045)" into jb-mr1-dev

Commit:
merge in jb-mr1-release history after reset to jb-mr1-dev

mikejurka committed:
Merge "Update custom cling params" into jb-mr1-dev

Commit:
Remove unnecessary allocations during draw
Bug #7327799

Change-Id: Ic3a18fb9d850f2c692dcb187a206c61dfa8108a4

Commit:
Merge "Import translations. DO NOT MERGE" into jb-mr1-dev

Commit:
Revert "Import translations. DO NOT MERGE"

broken build

This reverts commit 2826bc4fb6ec196eb72be7b9ef569996e2ff0dd4

Change-Id: I232f8abdf1fdb592ba76216fc82d836294ca4e09

Commit:
Merge "Revert "Import translations. DO NOT MERGE"" into jb-mr1-dev

mikejurka committed:
Add entries for string resources that appear in clings

Commit:
merge in jb-mr1-release history after reset to jb-mr1-dev

mikejurka committed:
Merge "Add entries for string resources that appear in clings" into jb-mr1-dev

Commit:
Merge "Remove unnecessary allocations during draw Bug #7327799" into jb-mr1-dev

Commit:
merge in jb-mr1-release history after reset to jb-mr1-dev

Commit:
Let's enhance

Add mipmap-* launcher icon assets. Removes equivalent icons from
drawable directories and fixes up the reference to them in a layout.

Bug: 6718166
Change-Id: Ic97e3208d0b47523c8fabef949b2b06825e2fa8f

Commit:
merge in jb-mr1-release history after reset to jb-mr1-dev

cm-gerrit commented:
Hi there;

This is an automated reply, to let you know CyanogenMod doesn't merge pull
requests submitted directly to our github.com repositories.

You can submit your change requests through our public review system,
located at http://review.cyanogenmod.com/

That review system is an installation of Google's Gerrit code review.
If you've never interacted with Gerrit before, we also suggest you take
a look at http://wiki.cyanogenmod.com/wiki/Howto:_Gerrit

Thanks!

On Tue, 16 Oct 2012 10:33:15 -0700 you wrote:

> This adds 2 preferences for the hotseat (Launcher main bar)
> - Choose the number of icons on the hotseat
> - Choose the place of the applications drawer button
> - Translations in English + French
> 
> This is usefull for big screen devices.
> You can set from 1 to 9 buttons on the hotseat.
> You can set the app button where you want.
> Default values  are keeped unchanged so there is no pb with first screen
> tuto + no pb with default shortcuts.
> 
> You can merge this Pull Request by running:
> 
>   git pull https://github.com/CmoaToto/android_packages_apps_Trebuchet jellybean
> 
> Or you can view, comment on it, or merge it online at:
> 
>   https://github.com/CyanogenMod/android_packages_apps_Trebuchet/pull/16
> 
> -- Commit Summary --
> - Trebuchet : Add 2 Hotseat preferences
> 
> -- File Changes --
> 
> M res/values-fr/strings.xml (4)
> M res/values-sw600dp/config.xml (2)
> M res/values/config.xml (2)
> M res/values/strings.xml (4)
> M res/xml/preferences.xml (13)
> M src/com/cyanogenmod/trebuchet/Hotseat.java (23)
> M src/com/cyanogenmod/trebuchet/Launcher.java (2)
> M src/com/cyanogenmod/trebuchet/preference/PreferencesProvider.java (9)
> 
> -- Patch Links --
> 
> https://github.com/CyanogenMod/android_packages_apps_Trebuchet/pull/16.patch
> https://github.com/CyanogenMod/android_packages_apps_Trebuchet/pull/16.diff
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/CyanogenMod/android_packages_apps_Trebuchet/pull/16

Commit:
Add missing layout attributes
Bug #7334849

Change-Id: Iae01cf5c587716951dee849e9c4116717d137d8f

CmoaToto committed:
Trebuchet : Add a pref to set hotseat icon's number

This adds a preference for the hotseat (Launcher main bar/dock)

- Choose the number of icons on the hotseat
- Choose the place of the applications drawer button by drag'n'drop

- Translations in English + French

This is usefull for big screen devices.
You can set from 1 to 9 buttons on the hotseat.
You can set the app button where you want by drag'n'drop.
Default values are keeped unchanged so there is no pb with first screen tuto + no pb with default shortcuts.

Demo video : https://www.dropbox.com/s/2wamj9yr1s3woxl/Trebuchet-hotseat-icons.mp4

Demo apk   : https://www.dropbox.com/s/psqab2gyivcyo3d/CmoaToto%20Trebuchet.apk

PatchSet 1 : initial commit

PatchSet 2 : coding style

PatchSet 3 : - Remove prefs for tablet
             - AllApps Button move by Drag'n'Drop (instead of pref)

PatchSet 4 : searchbar & deletebar don't dis/appear while dragging AllApps button)

PatchSet 5 : dropOnFlingToDeleteTarget disabled for AllApps button

PatchSet 6 : - Fix shortcuts errors in database
             - Set docks pref disabled for large screen programmatically
             - Fix english strings/french trads

Change-Id: I21f6a4610ed16c8780ccf6e2acffcf6e07511457

Conflicts:
	src/com/cyanogenmod/trebuchet/Hotseat.java
	src/com/cyanogenmod/trebuchet/LauncherModel.java
	src/com/cyanogenmod/trebuchet/Workspace.java

alanv committed:
Send correctly populated Accessibility event on Launcher state change.

Bug: 7362253
Change-Id: I72811292e8091a21866cb4599d37a71eb475b1d5

Commit:
merge in jb-mr1-release history after reset to jb-mr1-dev

Commit:
Bring dummy cling strings back as 'strings'

The translations import recognizes 'string', 'string-array' and
'plurals' as valid items to parse, all others are ignored.  This is the
reason, why these strings have not been imported back into the codebase.
Making the string declaration same as when it was removed in cl/236665

Change-Id: I6d53ba9d6267706834cb66e9ca99a1335aa01d14
Bug: 7203884

scottamain committed:
remove old fashioned dream, no longer supported

Change-Id: I87370e8dc7f3159182dac9f650223ce010d6286b

Commit:
merge in jb-mr1-release history after reset to jb-mr1-dev

mikejurka committed:
Fix NPE

Bug: 7371837

mikejurka committed:
Get rid of explicit GC requests

Bug: 7370913

mikejurka committed:
Fix NPE

Bug: 7356682
Change-Id: Ic6522da1ec6ac0d57f1565cef9c43949f2d7fd70

mikejurka committed:
Merge "Fix NPE" into jb-mr1-dev

Commit:
Merge "Bring dummy cling strings back as 'strings'" into jb-mr1-dev

Commit:
Adding width / height to cling text style

Change-Id: Ia39b049fe414edf926c221baed3b1223199ce2a7

sganov committed:
Merge "Send correctly populated Accessibility event on Launcher state change." into jb-mr1-dev

Commit:
merge in jb-mr1-release history after reset to jb-mr1-dev

yandexx committed:
Update Russian Translation-Trebuchet-CM10 (typo)

Change-Id: Ib9cebb36730055c8826b988bde09b82e566c7e9f

chadouming committed:
Fix page_hover_left_holo.9.png on all dpi

The 9patch file was wrong, instead of a black border
it was a black + alpha border so you would see the
black border instead of the desired effect.

Change-Id: I6f3c0a7271db11652a9dd1626f4563b660fcdc61

Commit:
Import translations. DO NOT MERGE

Change-Id: Ibc8c1dcd857312d4d6df39c18cc32708dfcaea80
Auto-generated-cl: translation import

Commit:
merge in jb-mr1-release history after reset to jb-mr1-dev

mikejurka committed:
Merge "Get rid of explicit GC requests" into jb-mr1-dev

Commit:
merge in jb-mr1-release history after reset to jb-mr1-dev

Commit:
Delay ACTION_MAIN processing when not focused.

This change puts a 350 ms delay before processing the ACTION_MAIN intent when
the window is not focused.  This makes the Launcher animation take place after
the GPU-intensive portion of the window shade hide animation, resulting in
smoother animations.

Bug: 7401764
Change-Id: I574f2ba398fcf38b423413f802673bbd4d983cd2

chethaase committed:
Resize wallpaper assets to avoid runtime resizing

Issue #7352961 Wallpaper edge is cut-off while scrolling through home page

Change-Id: I4bed3d4445fdcc2e2effd846f074bf7ef6dd532a

ygorigor committed:
Trebuchet: updated romanian translation

Change-Id: Idc72ef89dc351d18a9df4ea874828176d0a69a9c
Signed-off-by: Igor <igor_sorocean@hotmail.com>

chethaase committed:
Merge "Resize wallpaper assets to avoid runtime resizing" into jb-mr1-dev

Commit:
Merge "Delay ACTION_MAIN processing when not focused." into jb-mr1-dev

Commit:
merge in jb-mr1-release history after reset to jb-mr1-dev

chethaase committed:
Cap launcher swipe animations to max duration

large screens take too long swiping between screens. Cap the duration to
some max value to make things snappier.

Change-Id: Iee967c1b00b4962a0490fd0e20d48d7d8363dd2f

chethaase committed:
Cap launcher swipe animations to max duration

large screens take too long swiping between screens. Cap the duration to
some max value to make things snappier.

Change-Id: Iee967c1b00b4962a0490fd0e20d48d7d8363dd2f

Commit:
Merge "Cap launcher swipe animations to max duration" into jb-mr1-dev

CmoaToto committed:
Set Trebuchet as standalone apk (cyanogenmod became cmoatoto)

CmoaToto committed:
Trebuchet : Add a pref to set hotseat icon's number

This adds a preference for the hotseat (Launcher main bar/dock)

- Choose the number of icons on the hotseat
- Choose the place of the applications drawer button by drag'n'drop

- Translations in English + French

This is usefull for big screen devices.
You can set from 1 to 9 buttons on the hotseat.
You can set the app button where you want by drag'n'drop.
Default values are keeped unchanged so there is no pb with first screen tuto + no pb with default shortcuts.

PatchSet 1 : initial commit

PatchSet 2 : coding style

PatchSet 3 : - Remove prefs for tablet
             - AllApps Button move by Drag'n'Drop (instead of pref)

PatchSet 4 : searchbar & deletebar don't dis/appear while dragging AllApps button)

PatchSet 5 : dropOnFlingToDeleteTarget disabled for AllApps button

PatchSet 6 : Add German L10n

CmoaToto committed:
Merge remote-tracking branch 'upstream/jellybean' into jellybean

CmoaToto committed:
Trebuchet: Added French translations

chethaase committed:
Stop animating All Apps during transition to Home

One of the sources of jank in launcher is during the All Apps -> Home
transition. specifically, if the user has started a fling operation (where
we are animating between pages in All Apps) and then hits the Home button,
we continue the fling animation while also doing the transition to Home
scale/fade animations. This causes a lot of work for launcher, particularly because
the fling animation is causing the All Apps layer to get recreated on every frame.

The fix is to simply pause the fling animation, then snap to its end state when the
animation to Home is complete. We also need to pause/snap the scroll indicator animation,
because it's fading animation causes the same layer-recreation jank that the fling
itself causes.

Issue #7387124 Home <-> All Apps transition animation is janky while flinging

Change-Id: Icbcaf2d5b3b2f6ce8fd7419419d258248aa1475b

maxwen committed:
Fix for NPE in showAllAppsCling

Added check for getTabHost()

E/AndroidRuntime( 1880): FATAL EXCEPTION: main
E/AndroidRuntime( 1880): java.lang.NullPointerException
E/AndroidRuntime( 1880): 	at com.cyanogenmod.trebuchet.AppsCustomizePagedView.showAllAppsCling(AppsCustomizePagedView.java:575)
E/AndroidRuntime( 1880): 	at com.cyanogenmod.trebuchet.AppsCustomizeTabHost.onLauncherTransitionEnd(AppsCustomizeTabHost.java:455)
E/AndroidRuntime( 1880): 	at com.cyanogenmod.trebuchet.Launcher.dispatchOnLauncherTransitionEnd(Launcher.java:2401)
E/AndroidRuntime( 1880): 	at com.cyanogenmod.trebuchet.Launcher.access$2100(Launcher.java:120)
E/AndroidRuntime( 1880): 	at com.cyanogenmod.trebuchet.Launcher$14.onAnimationEnd(Launcher.java:2519)
E/AndroidRuntime( 1880): 	at android.animation.AnimatorSet$AnimatorSetListener.onAnimationEnd(AnimatorSet.java:765)
E/AndroidRuntime( 1880): 	at com.cyanogenmod.trebuchet.LauncherViewPropertyAnimator.onAnimationEnd(LauncherViewPropertyAnimator.java:111)

Change-Id: I4aa0b4bacc03fb7aa87029d3e9406570bc82830f

chethaase committed:
Merge "Stop animating All Apps during transition to Home" into jb-mr1-dev

Commit:
merge in jb-mr1-release history after reset to jb-mr1-dev

Commit:
Import translations. DO NOT MERGE

Change-Id: I1fcec61bfde318f1dca85130a25d0bb75491d8ab
Auto-generated-cl: translation import

Commit:
merge in jb-mr1-release history after reset to jb-mr1-dev

Commit:
Import translations. DO NOT MERGE

Change-Id: Ia8ddaa0b2fb9877eb3e6d2f761f66f756065ae2d
Auto-generated-cl: translation import

DelphinPETER committed:
Trebucher : Added missing French Translation

Change-Id: I91fd42322fce3781dcd029b9df4a5e9fc7268ff4

Commit:
merge in jb-mr1-release history after reset to jb-mr1-dev

CmoaToto committed:
Trebuchet : Add a pref to set hotseat icon's number

PatchSet 6 : Fix shortcuts errors in database + set docks pref disabled for large screen programmatically + fix strings/french trads

CmoaToto committed:
Fix Merge conflicts

CmoaToto committed:
Trebucher : Added missing French Translation

PatchSet 1 : Add a correction, thx to Florent Imfloflo

Change-Id: I17854dbb9cb6507e6797357fb65ae9e43db02142

Commit:
Import translations. DO NOT MERGE

Change-Id: Ia8f60a9599833d4eccc6f332af67b4a71ba12a51
Auto-generated-cl: translation import

Commit:
merge in jb-mr1-release history after reset to jb-mr1-dev

rmcc committed:
Fix post-merge breakage from the new 4.2 stuff

rmcc committed:
Merge commit 'mr1-rebase' into mr1-staging

rmcc committed:
Underp folder scaling

Tortel committed:
Fix package mismatch

Change-Id: I97ee13ad3507d083374791f58b1bf8c727d7c793

burnsra committed:
Fix desktop alignment when Search Bar is hidden

When search bar is hidden all items shift to the left too far.

Screenshots for reference:
Search Bar Visible
https://docs.google.com/open?id=0B9gh9VnHidOdLVFNTUtKbzdWNlE

Search Bar Hidden (Widget is too far to the left)
https://docs.google.com/open?id=0B9gh9VnHidOdZ1dXVVFONDVrVHc

Change-Id: I2a37b88ae4cb082143c49371b51a1578660e9321

nebkat committed:
Merge "Fix desktop alignment when Search Bar is hidden" into mr1-staging

Commit:
Wallpaper Hack

Normally the wallpaper is drawn by the system in a separate window.
This means that the wallpaper can often be drawn out of sync with
the launcher. This hack gets the wallpaper bitmap and draws it
manually (in sync with the rest of the launcher). This increases
performance greatly.

Change-Id: I6aaf9c574cabcf2ee0bfeef1076140421108331d

Commit:
Wallpaper Hack: Default to false

Change-Id: I777fa2b8b6d4f0bc496b618977dfe4bea43ad211

Commit:
LauncherModel: Fix possible NPE

Change-Id: I97a48c62a4b111b2b6a20093e8b7346fec9ae3ea

Commit:
oops

Change-Id: I9d8c3860851ea3eda632c8059d5a50a785722c4e

Commit:
Workspace: Update wallpaper offsets to match hack for recent apps window

Change-Id: I16ad78f059ca4689b54b9e6d3806d5d89157d2f2

Commit:
Workspace: Don't refresh wallpaper visibility on live wallpapers

Change-Id: I82872980ec4aa795ab14a1bdad90069701a983ed

Commit:
Workspace: Make sure wallpaper gets redrawn to avoid ghost wallpapers

Change-Id: I5c8a699b810e0e39932d1706c87fe1be7335522f

rmcc committed:
Fix various cell size/placement failures

Restore the original L2 Shortcut/widget sizing logic, adapted to
the device's (default or user-configured) grid size at the time of
cell creation. Using onMeasure + padding doesn't look as good, and
renders different results depending on the device density.

This results in the same workspace peripheral padding as L2,
regardless of grid size.

Change-Id: Ic220b63df31322365d48a7fb593d7d9a170057f3

rmcc committed:
This is causing a lot of weirdness on tablets' hotseats, + mild weirdness (such as roaming AllApps icon) on phones

Revert "Trebuchet : Add a pref to set hotseat icon's number"

This reverts commit e74a54343ce7896b5005154e8bcbc7d5f06c7808

Commit:
Workspace: Quad interpolator

Change-Id: I995a311c60ed775e818c4b17393964111214ff7d

Commit:
Fix workspace padding preference

Change-Id: If10000284142bac54d22a58daa2e4f7fd0135a90

Commit:
Workspace: Don't set wallpaper visibility with hack

Change-Id: Ib3c21db1535a0a3c8f41a9cddfa21ade0ecebe84

Commit:
Workspace: Remove wallpaper shadow background

Change-Id: Idfaeb58aae5150101bcf191d37103a5f5645b27a

Commit:
Workspace: Show outlines

Change-Id: I7345092146583906a0e597774d84790e0c471268

Commit:
Workspace: Fade in adjacent screens

Change-Id: Ifeb3bc1e7e75782de2cb20b646defd599cc55406

Commit:
Update Russian Translation-Trebuchet-CM10.1

Update Russian Translation-Trebuchet-CM10.1

Commit:
Preferences: Use default grid size params

Change-Id: I965132939eea1d97f6ca02d4152bfab2e4b436f4

Commit:
Workspace: Fix missing show outlines

Change-Id: I893002dd7aa1c556c6d6b712ce042e5d9c5eac7e

elektroschmock committed:
German: Update translation

Patch set 1 - Add translations for:
              * Wallpaper Hack
              * Homescreen outline
              * Homescreen fading

Patch set 2 - Remove 'application_name' in ca, fa, sr

Patch set 3 - Add transition effects
              make some arrays nontranslatable

Change-Id: Ib7f71650b315f7568dfb63e8bd1faf611fff70d8

Commit:
Transition Effects

Change-Id: I9191cad9ad9950144fffd75e5140ae11759eecc8

Commit:
Revert "Workspace: Don't set wallpaper visibility with hack"

This reverts commit a61c75cffbd5b3fb630398586da3844752f61f5d.

Commit:
Workspace: Fix wallpaper offset on recents list

Change-Id: I2c81bd5f3395c2da1a74d190ba2a3f6ee1da5de6

Commit:
Preferences: Use correct default padding values

Change-Id: I2340d3b043fc4e80a49ec1623241c9bbc6b62dbe

Commit:
Workspace: Stretch screens

Change-Id: I5b587ec8e90a45e5e1df6112299ab298c0f907be

Commit:
PagedView: Don't force draw children without checking whether they should be drawn

Change-Id: I93926d75426ccfe139ce05cf6561ced264e9e807

Commit:
AppsCustomizePagedView: Flip transition effect camera distance

Change-Id: I7163366ebe0b797270b8a54369aa9f6ffda1453d

Commit:
PagedView: Vertical scroll indicator

Change-Id: Ia35fdd69dccacbd7dbeb2749b2c2fbf51973cb64

Commit:
Update Russian Translation-Trebuchet-CM10.1

CmoaToto committed:
Fix Merge conflicts

CmoaToto committed:
Merge branch 'cm-10.1' of git://github.com/CyanogenMod/android_packages_apps_Trebuchet into jellybean

Conflicts fixed:
	AndroidManifest.xml
	res/layout-land/workspace_cling.xml
	res/layout-port/search_bar.xml
	res/layout-port/workspace_cling.xml
	res/layout-sw720dp-land/application.xml
	res/layout-sw720dp-port/application.xml
	res/layout-sw720dp-port/workspace_cling.xml
	res/layout-sw720dp/search_bar.xml
	res/layout-sw720dp/workspace_cling.xml
	res/values-fr/strings.xml
	res/xml/preferences.xml
	src/com/cmoatoto/trebuchet/AllAppsView.java
	src/com/cmoatoto/trebuchet/AppsCustomizePagedView.java
	src/com/cmoatoto/trebuchet/CellLayout.java
	src/com/cmoatoto/trebuchet/Cling.java
	src/com/cmoatoto/trebuchet/Hotseat.java
	src/com/cmoatoto/trebuchet/Launcher.java
	src/com/cmoatoto/trebuchet/LauncherModel.java
	src/com/cmoatoto/trebuchet/PagedViewWidgetImageView.java
	src/com/cmoatoto/trebuchet/SearchDropTargetBar.java
	src/com/cmoatoto/trebuchet/Workspace.java

eljef commented:
Please submit through gerrit. http://review.cyanogenmod.org
